### ISSUE TITLE
Password strength

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-websites#1461]
So apparently the operations department have decided to implement a password policy for joomla.org properties that **requires** the use of at least one special character.

PLEASE reconsider this policy as it reduces security instead of enhancing it

The National Institute of Standards and Technology (NIST) specifically state the following in the NIST cybersecurity framework (https://pages.nist.gov/800-63-3/sp800-63b.html#sec5)

> Verifiers SHOULD NOT impose other composition rules (e.g., requiring mixtures of different character types or prohibiting consecutively repeated characters) for memorized secrets. Verifiers SHOULD NOT require memorized secrets to be changed arbitrarily (e.g., periodically). However, verifiers SHALL force a change if there is evidence of compromise of the authenticator.

As a result  of this advice neither the Google Chrome password manager or Firefox lockwise will always generate passwords with special characters nor can they be configured to do so. The same is true for several commercial password managers.

Please do NOT blindly close and lock this because you have the power to do so and you disagree. Please respect the request for the department to reconsider the policy which I am sure was made with the best of intentions but was made in absence of the facts and NIST recommendations.

For a low tech explanation watch the beginnings of this presentation at USENIX by Dropbox
https://www.usenix.org/conference/usenixsecurity16/technical-sessions/presentation/wheeler

**Comment by conconnl** [comment]
@HLeithner @zero-24 & @JazParkyn 
As we asked your advice first, before we wrote the proposal and asked all webmasters feedback, while most didn't had any. I would like to ask you again to reevaluate your advice.
I will cancel the motion for today, to not have to repeat the process again.

@sanderpotjer & @roland-d I would also still like your feedback on it, as the bases for the policy is coming from the Identity (IDP) system built by you together with Compliance. Changes in the proposal means changes in the Identity site.

**Comment by brianteeman** [comment]
PS Using the dropbox library would be a real strength test https://github.com/dropbox/zxcvbn

**Comment by conconnl** [comment]
> PS Using the dropbox library would be a real strength test https://github.com/dropbox/zxcvbn

It's not in Joomla as a feature... Maybe you can propose a PR for J4!

**Comment by brianteeman** [comment]
@conconnl I did - no one other than @jeckodevelopment commented despite specifically asking JSST and maintainers

**Comment by conconnl** [comment]
That’s to bad! What is the number?

**Comment by HLeithner** [comment]
There are always different opinions, my advice was to use long passwords 10+ and with lowest restrictions possible. But I can accept more restrict password requirements. (Even if I failed on them yesterday ;-)

**Comment by brianteeman** [comment]
@HLeithner was correct recommending no restrictions.

**Comment by roland-d** [comment]
So I watched the full video Brian linked basically it says, the zxcvbn checker is more accurate. Great, so why not use it?

> So apparently the operations department have decided to implement a password policy for joomla.org properties that requires the use of at least one special character.

> PLEASE reconsider this policy as it reduces security instead of enhancing it

That I don't see in the video however I tried out the online demo at https://lowe.github.io/tryzxcvbn/ and adding a special character does seem to add to the difficulty of guessing the password.

As for the NIST standard, in the video they are not too excited about it :)

Long story short, if a company like Dropbox reckons 6 characters minimum is enough, why do it any different? 

Google has an 8 character minimum (source: https://support.google.com/accounts/answer/32040?hl=en ) 

So let's keep it simple :)

P.s. I cannot find the PR/Issue that proposes to implement zxcvbn

**Comment by brianteeman** [comment]
Thanks for taking the time to watch the video. It is appreciated that you did and didnt just ignore this issue and close it as happened yesterday.

> As for the NIST standard, in the video they are not too excited about it :)

The video was recorded in 2016. At that time NIST did recommend using special characters. In 2019, perhaps as a result of this, they reversed that and today they do NOT recommend it. So the speakers comments are referring to the **old** recommendations

> P.s. I cannot find the PR/Issue that proposes to implement zxcvbn

https://github.com/joomla/joomla-cms/issues/25914

**Comment by Radek-Suski** [comment]
To be honest if this requirement is breaking using password managers, which is definitely recommended, we should reconsider the requirement.

**Comment by brianteeman** [comment]
and it would probably take a real dev only a few hours to replace the current misnamed password strength meter with zxcvbn https://css-tricks.com/password-strength-meter/

**Comment by conconnl** [comment]
I'm done with this way of communicating, pointing fingers, making assumptions etc.
The facts are very simple.

* We didn't and still don't have requirements and every webmaster can still do what they like with their website for this point.
* Feedback was requested to 3 experienced people (see above) and their feedback was converted to a table.
* The table was shared with the same 3 people to get confirmation if everything was OK, They all confirmed it.
* Feedback was requested (including reminders) to the whole Webmasters team, one person took the effort to correct some English, another person just to ask a question about versioning. The rest involved didn't bother to do any feedback, including leadership!

For me it doesn't matter at all what the result will be, as long as there will be a standard for all joomla websites.

As long as the responsible people for the websites, don't bother to work together on improving our platform, standardize things and communicate (like providing feedback when asked, or even unasked). I'm not going put myself in the same place as Michael has for many years, doing everything himself, getting a lot of shit in his face even from people who should be his partner (especially people who do everything publicly, GitHub, Social Media, etc and don't have the respect and decency to contact someone privately and respectful through mail and glip)

**Comment by brianteeman** [comment]
You were the person who closed and locked the previous valid report - still awaiting your apology for that mistake. 

A mistake was made. I pointed it out. 
What happened then was you said that the JSST had mandated the policy (you've now denied that) and locked the issue.
What should have happened was what is happening now.

**Comment by conconnl** [comment]
Sorry I forgot you are always right and everybody else is wrong.
My mistake, I have to keep reminding myself that you are perfect.
That your way off communicating is the correct way no matter if you are disrespectful on so many occasions.

I'm so sorry that I'm not like you and involve myself in everything happening in the organization, and that I don't know everything like you do and sorry that I ask other people for feedback (if they bother to give it) and they also approve it, but that it isn't what you want.
From now one we should ask everything to you, then you have nothing to complain anymore, maybe that would make your live better.

**Comment by jeckodevelopment** [comment]
Can we have a more collaborative approach?
Decisions were not made by one person, but have been discussed within the team and accepted.
If they are technically wrong, then can be discussed again and eventually amended, but please, let's express opinions in a collaborative way. Thanks.

**Comment by brianteeman** [comment]
I give up. You can stop with the attack dogs now.

**Comment by conconnl** [comment]
The none-approved by motion defined rules by Webmasters team are now approved.
The policy consist of more then just password rules.
The rules for passwords are defined by the webmasters in the following manner.

<img width="906" alt="image" src="https://user-images.githubusercontent.com/11913918/85030974-66634980-b17e-11ea-83a2-16110de94ca0.png">

Now that it's an official policy all sites must and will be reconfigured to comply with them and use the same rules everywhere.

**Comment by brianteeman** [comment]
I accept your apology for your mistaken comments earlier
